### DISCUSSION TITLE
SSH log successful login by publickey too

### DISCUSSION THREAD

**Discussion by chrisandchris** [DigitalRuby/IPBan#196] [Ideas]
Hi there,

I have a question/input about the defaults for SSH log file parsing. The default only matches "Accepted password [...]", however this does not match any authentication with publickey (or other authentication method). See: https://github.com/DigitalRuby/IPBan/blob/master/IPBanCore/ipban.config#L58

E.g. the message when authentication with a publickey looks like this:

```
Accepted publickey for [user] from [ip] port [port] ssh2: [...]
```

I suggest either extending the regex in ipban.config to also match publickey auth or using a wildcard (\w) to match any authentication method. Or is there any disadvantage I'm not seeing? Maybe the same applies to failed logins (using an invalid/wrong key pair).

Best regards,
Christian

**Comment by jjxtra** [comment]
Can you post the actual log file line? You can mask out the user and ip. I'll add this to the regex.

**Comment by chrisandchris** [comment]
Sure, I'll take an example from https://www.elastic.co/de/blog/grokking-the-linux-authorization-logs which looks like mine (with different usernames / IPs obvsiouly):

```
Feb 20 21:54:44 localhost sshd[3402]: Accepted publickey for vagrant from 10.0.2.2 port 63673 ssh2: RSA 39:33:99:e9:a0:dc:f2:33:a3:e5:72:3b:7c:3a:56:84
Feb 21 00:13:35 localhost sshd[7483]: Accepted password for vagrant from 192.168.33.1 port 58803 ssh2
```

**Reply by jjxtra** [reply]
https://github.com/DigitalRuby/IPBan/commit/2ea93c528793e24f0affa37966874540a248601a

**Reply by chrisandchris** [reply]
It does not catch the 1st line, just the 2nd one (that was the original input): https://regex101.com/r/qR6cIU/1

**Comment by jjxtra** [comment]
Ok, I believe this if fixed in 1.7.2 which I refreshed today with commit https://github.com/DigitalRuby/IPBan/commit/69a141555e3c1af10dca299d9bc452ff4720a29a

Let me know if you see the same
